private/bufpkg/bufimage/bufimage.go

```diff
@@ -129,6 +129,13 @@ func NewMultiImage(images ...Image) (Image, error) {
 // be treated as non-imports in the returned Image. The first non-import
 // version of a file will be used in the result.
 //
+// Note that the files within each image are expected to be de-duplicated
+// before being passed into this function. In the case that two images
+// consider the same file as a non-import, the first will be used. We purposefully
+// do not validate this condition here as a result of the --path filter, and
+// instead validate it as part of the ParserAccessorHandler when each Image
+// is constructed.
```

Comment:
I investigated what it would look like to include this validation in each `ModuleFileSet` independently, but realized that we get it for free in the `ParserAccessorHandler` after the fact.

We _could_ add validation that acts upon all of the `ModuleFileSets` together, but it doesn't buy us anything in practice. Having multiple `ModuleFileSets` is only a detail for workspaces, and every `Module` will already be contained in each of these `ModuleFileSets` (as either the primary module, or a dependency). If each `ModuleFileSet` is valid and de-duplicated within itself, then the same holds for its relationship between the rest.

If we were to go this route, we would probably want to merge `ModuleFileSets` into one, but this is something that we should discuss separately since there are other implications.